lib/nixd/include/nixd/Server.h

```diff
@@ -36,18 +41,22 @@ bool fromJSON(const llvm::json::Value &Params, InstallableConfigurationItem &R,
 class Server : public lspserver::LSPServer {
 
   EvalDraftStore DraftMgr;
-  boost::asio::thread_pool Pool;
+  boost::asio::thread_pool Pool = boost::asio::thread_pool(1);
```

Comment:
Currently we cannot handle race conditions in nix GC, and looks like parallel evaluation crashes the server.